cpp/bench/ann/src/common/benchmark.hpp

```diff
@@ -135,6 +135,12 @@ void bench_build(::benchmark::State& state,
     }
   }
 
+  if (index.algo == "diskann_ssd") {
+    make_sure_parent_dir_exists(index.file);
+    index.build_param["dataset_file"]  = dataset->base_filename();
+    index.build_param["path_to_index"] = index.file;
+  }
+
```

Comment:
I think you shouldn't need `path_to_index` parameters, because the index member `file` is accessible from within the index in your algorithm-specific code.
If you need the dataset file path during index build,  you can either ingest it as a build param in the `conf.hpp`, similar to how we inject some search parameters in [parse_index(...)](https://github.com/rapidsai/cuvs/blob/11ecbb2a263e3f6591d0a2390453ca0356b845c0/cpp/bench/ann/src/common/conf.hpp#L106-L141), or create one more member in the `configuration::index` - all in `conf.hpp`; I think it's ok to have it available to all algorithm at that point.

Reply:
Unfortunately, the `cuvs::bench::configuration::index` object's attributes are not directly passed down to the specific algorithm's wrapper class. So we would need the `path_to_index`. The algorithm specific structs use only the parameters that are explicitly passed down from the configuration object.

Reply:
My question is: CAN we pass the `cuvs::bench::configuration::Index` down into the wrappers? Or rather, why shouldn't we?

Reply:
Fixed via configuration singleton